### PR TITLE
Drain state implementation

### PR DESCRIPTION
**Pull request by adrianchiris**
Refer to individual commits for context.

### CODE REVIEW

File: internal/drain/request.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/drain"
+
+	maintenancev1 "github.com/Mellanox/maintenance-operator/api/v1alpha1"
+	"github.com/Mellanox/maintenance-operator/internal/log"
+)
+
+// DrainRequestUIDFromNodeMaintenance returns DrainRequest UID of given NodeMaintenance object.
+func DrainRequestUIDFromNodeMaintenance(nm *maintenancev1.NodeMaintenance) metav1types.UID {
+	return metav1types.UID(fmt.Sprintf("%s/%s:%s", nm.Namespace, nm.Name, nm.UID))
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply internal/drain/request.go:39]: makes sense.

so shall i just use `string` ? or define type alias for string ?
the former will save me  type casting in code :)

File: internal/controller/nodemaintenance_controller.go [hunk 2]
```diff
@@ -353,6 +474,40 @@ func (r *NodeMaintenanceReconciler) handleWaitPodCompletionState(ctx context.Con
...
+		}
+
+		nm.Status.Drain = &maintenancev1.DrainStatus{
+			TotalPods:    uint32(len(podsOnNode.Items)),
+			EvictionPods: uint32(len(ds.PodsToDelete)),
+		}
+	}
+
+	removedPods := utils.MaxInt(int(nm.Status.Drain.EvictionPods)-len(ds.PodsToDelete), 0)
+	nm.Status.Drain.DrainProgress = uint32(float32(removedPods) / float32(nm.Status.Drain.EvictionPods) * 100)
```
Comment (adrianchiris) [inline internal/controller/nodemaintenance_controller.go:500]: should handle case where EvictionPods is 0

else we divide by zero which is not nice.

File: .mockery.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+inpackage: False
+testonly: False
+with-expecter: True
+keeptree: True
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply .mockery.yaml:4]: done.

File: internal/controller/nodemaintenance_controller.go (hunk 2 above)
Comment (adrianchiris) [reply internal/controller/nodemaintenance_controller.go:500]: fixed.

File: internal/drain/request.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianchiris) [reply internal/drain/request.go:39]: went with string, LMK if you prefer something else.

**Review by ykulazhenkov — APPROVED** [review]
LGTM

**Pull request merged**